internal/install/segment/client.go

```diff
@@ -0,0 +1,95 @@
+package segment
+
+import (
+	"encoding/json"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+type EventType string
+
+var EventTypes = struct {
+	InstallStarted          EventType
+	AccountIDMissing        EventType
+	APIKeyMissing           EventType
+	RegionMissing           EventType
+	UnableToConnect         EventType
+	UnableToFetchLicenseKey EventType
+	LicenseKeyFetchedOk     EventType
+	UnableToOverrideClient  EventType
+}{
+	InstallStarted:          "InstallStarted",
+	AccountIDMissing:        "AccountIDMissing",
+	APIKeyMissing:           "APIKeyMissing",
+	RegionMissing:           "RegionMissing",
+	UnableToConnect:         "UnableToConnect",
+	UnableToFetchLicenseKey: "UnableToFetchLicenseKey",
+	LicenseKeyFetchedOk:     "LicenseKeyFetchedOk",
+	UnableToOverrideClient:  "UnableToOverrideClient",
+}
+
+type Segment struct {
+	analytics.Client
+}
+
+func New(writeKey string) *Segment {
+	if writeKey == "" {
+		log.Debug("segment: write key is empty, cannot write to segment")
+		return nil
+	}
+
+	client := analytics.New(writeKey)
+	return &Segment{client}
+}
+
+func (client *Segment) Track(accountID int, event Event) {
+
+	if client == nil {
+		return
+	}
+
+	properties := toMap(event)
+
+	properties["category"] = "newrelic-cli"
+	properties["accountId"] = accountID
+
+	err := client.Enqueue(analytics.Track{
+		UserId:     fmt.Sprintf("%d", accountID),
+		Event:      "VirtuosoCLIInstall",
```

Comment:
@NRhzhao check IES, this isn't right. the Event field should be the eventName.
The method signature should allow an optional interface{ } to get additional properties.

Reply:
event name is the table name, in IES there are two event names that correspond to the two tables we have.
I figure for CLI, we want only one table so I hardcode the one table name there.
In IES, the event is a interface type, so it can be any type and gets converted to a dictionary. 
Since we only logging event here, I thought to keep it as a concrete type instead.

![image](https://user-images.githubusercontent.com/94071514/236579533-aa930c70-16f9-4c3f-bd75-8173970595d5.png)